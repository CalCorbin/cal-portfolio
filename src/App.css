.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.construction-font {
  background-color: #028090;
  color: #fff;
  font-size: calc(10px + 2vmin);
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-family: "Lucida Console", Courier, monospace;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
