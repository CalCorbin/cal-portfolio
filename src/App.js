import React, { useMemo } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
import Firebase, { FirebaseContext } from './components/Firebase';
import ROUTES from './constants/routes';
import Landing from './pages/Landing';
import CatChat from './pages/CatChat';
import Resume from './pages/Resume';
import SpaceX from './pages/SpaceX';
import HexClock from './pages/HexClock';

const App = function () {
  const client = new ApolloClient({
    uri: 'https://api.spacex.land/graphql/',
    cache: new InMemoryCache(),
  });

  const newFireBase = useMemo(() => new Firebase(), []);

  return (
    <FirebaseContext.Provider value={newFireBase}>
      <ApolloProvider client={client}>
        <div data-testid="app-container">
          <BrowserRouter>
            <Routes>
              <Route path={ROUTES.RESUME} element={<Resume />} />
              <Route path={ROUTES.SPACEX} element={<SpaceX />} />
              <Route path={ROUTES.CATCHAT} element={<CatChat />} />
              <Route path={ROUTES.HEXCLOCK} element={<HexClock />} />
              <Route path={ROUTES.LANDING} element={<Landing />} />
            </Routes>
          </BrowserRouter>
        </div>
      </ApolloProvider>
    </FirebaseContext.Provider>
  );
};

export default App;
