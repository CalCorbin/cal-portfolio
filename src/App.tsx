import React, { useMemo } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
import ROUTES from './constants/routes';
import Landing from './pages/Landing/Landing';
import CatChat from './pages/CatChat/CatChat';
import SpaceX from './pages/SpaceX/SpaceX';
import HexClock from './pages/HexClock/HexClock';
import MagicNumber from './pages/MagicNumber/MagicNumber';
import ChicagoArt from './pages/ChicagoArt/ChicagoArt';
import HomeScreen from './components/TicTacToe/HomeScreen';
import MineSweeper from './components/MineSweeper/MineSweeper';

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import Firebase, { FirebaseContext } from './components/Firebase';

const App = () => {
  const client = new ApolloClient({
    uri: 'https://main--spacex-l4uc6p.apollographos.net/graphql',
    cache: new InMemoryCache(),
  });

  const newFireBase = useMemo(() => new Firebase(), []);
  const queryClient = new QueryClient();

  return (
    <FirebaseContext.Provider value={newFireBase}>
      <QueryClientProvider client={queryClient}>
        <ApolloProvider client={client}>
          <div data-testid="app-container">
            <BrowserRouter>
              <Routes>
                <Route path={ROUTES.SPACEX} element={<SpaceX />} />
                <Route path={ROUTES.CATCHAT} element={<CatChat />} />
                <Route path={ROUTES.HEXCLOCK} element={<HexClock />} />
                <Route path={ROUTES.LANDING} element={<Landing />} />
                <Route path={ROUTES.MAGICNUMBER} element={<MagicNumber />} />
                <Route path={ROUTES.CHICAGOART} element={<ChicagoArt />} />
                <Route path={ROUTES.TICTACTOE} element={<HomeScreen />} />
                <Route path={ROUTES.MINESWEEPER} element={<MineSweeper />} />
              </Routes>
            </BrowserRouter>
          </div>
        </ApolloProvider>
      </QueryClientProvider>
    </FirebaseContext.Provider>
  );
};

export default App;
