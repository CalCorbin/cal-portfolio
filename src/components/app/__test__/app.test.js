import React from 'react';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import App from '../index';

afterEach(cleanup);

it('should match the app snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
