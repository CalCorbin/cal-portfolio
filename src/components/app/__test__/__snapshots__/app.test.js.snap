// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the app snapshot 1`] = `
<div
  data-testid="app-container"
>
  <div
    data-testid="landing-route"
  >
    <div
      className="__landing-container"
      data-testid="landing-container"
    >
      <div
        data-testid="landing-image-header"
      >
        <div
          alt="hongkong"
          className="__landing-header-image"
          data-testid="header-image"
        />
        <div
          className="__landing-header-text"
          data-testid="landing-header-text"
        >
          cal corbin
          <br />
          <br />
          coder | creator
        </div>
      </div>
      <div
        data-testid="landing-navigation"
      >
        <div>
          <div
            className="__button-container"
            data-testid="nav-button"
          >
            <p>
              About
            </p>
          </div>
        </div>
        <div>
          <div
            className="__button-container"
            data-testid="nav-button"
          >
            <p>
              Work
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
